### PR TITLE
chore: fix golangci-lint `Files Exists` errors

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

ref: 
https://github.com/prometheus-operator/prometheus-operator/actions/runs/7457201435/job/20289014732
https://github.com/prometheus-operator/prometheus-operator/actions/runs/7481957028/job/20364568306?pr=6198

```
  /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/go.sum: Cannot open: File exists
  Error: /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/cmd/sloglint/main.go: Cannot open: File exists
  Error: /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/sloglint.go: Cannot open: File exists
  /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/.goreleaser.yml: Cannot open: File exists
  /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/go.mod: Cannot open: File exists
  /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/.golangci.yml: Cannot open: File exists
  /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/LICENSE: Cannot open: File exists
  /usr/bin/tar: ../../../go/pkg/mod/go-simpler.org/sloglint@v0.1.2/README.md: Cannot open: File exists
```

Solve the occasional instability problem of golangci-lint.

Solution: hotfix is to skip the pkg cache
https://github.com/golangci/golangci-lint-action/issues/244#issuecomment-884807976

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -57,7 +57,8 @@ jobs:
         uses: golangci/golangci-lint-action@v3.7.0
         with:
           version: v1.55.2
-          args: --timeout 10m0s
+          args: --timeout=10m0s --go=${{ env.golang-version }}
```
Comment (simonpasquier) [inline .github/workflows/checks.yaml:60]: is this change needed? If possible it should another PR to have a cleaner change history.

File: .github/workflows/checks.yaml [hunk 2]
```diff
@@ -57,7 +57,8 @@ jobs:
...
+          skip-pkg-cache: true
```
Comment (simonpasquier) [inline .github/workflows/checks.yaml:61]: Can you add a comment
```suggestion
          # Disable caching as a workaround for https://github.com/golangci/golangci-lint-action/issues/135.
          # The line can be removed once the golangci-lint issue is resolved.
          skip-pkg-cache: true
```

**Review by simonpasquier — COMMENTED** [review]
thanks!

File: .github/workflows/checks.yaml (hunk 2 above)
Comment (ArthurSens) [reply .github/workflows/checks.yaml:61]: Ideally, we don't want to skip the cache, but a fix upstream :P 

The comment will be a good reminder that we want to eventually enable cache again

File: .github/workflows/checks.yaml (hunk 1 above)
Comment (dongjiang1989) [reply .github/workflows/checks.yaml:60]: Got it.  I will add another PR

File: .github/workflows/checks.yaml (hunk 2 above)
Comment (dongjiang1989) [reply .github/workflows/checks.yaml:63]: Got it

Comment (dongjiang1989) [reply .github/workflows/checks.yaml:63]: Got it

File: .github/workflows/checks.yaml [hunk 3]
```diff
@@ -57,7 +57,10 @@ jobs:
         uses: golangci/golangci-lint-action@v3.7.0
         with:
           version: v1.55.2
-          args: --timeout 10m0s
+          args: --timeout=10m0s --go=${{ env.golang-version }}
```
Comment (dongjiang1989) [inline .github/workflows/checks.yaml:60]:
```suggestion
          args: --timeout 10m0s
```
